### ISSUE TITLE
Is there a Setting that will prevent new lines from having // when I press Enter?

### ISSUE THREAD

**Issue by gaveitatry02** [violentmonkey/violentmonkey#631]
Some of my lines are commented out with //.  If my cursor is at the end of those lines and I press Enter, each new line will start with a //.  Yes, you don't need to tell me that I can put the cursor at the end of non-commented lines before pressing Enter.  But even if I try extremely hard to train myself to remember and always do that, there is still a chance that I will make the mistake or forget and then end up with a bunch of new lines with // at the beginning of them.  The thing is, I NEVER EVER need new lines to start off commented.  So I was wondering if there is a Setting so that I can turn that off, or if the ViolentMonkey developer could add a NEW Setting that will let me toggle that off.  Thanks.

**Comment by tophf** [comment]
Yep, it's wrong.
I think the only cases when `//` should be inserted automatically are:
* pressing Enter in the middle of an existing `//` comment, not at the end
* maybe also when pressing Enter inside the metadata comment

**Comment by gaveitatry02** [comment]
I just want to 100% toggle it off.  If I want //, I can take less than one second to type //.  I am not saving time by having ViolentMonkey do it for me.

**Comment by tophf** [comment]
This is a problem in CodeMirror's addon and I've [submitted a fix there](https://github.com/codemirror/CodeMirror/pull/6022). It may take some time (a week, a month, or more) to get the new version published.

Meanwhile you can use a workaround: open devtools (F12 or Ctrl-Shift-i or Cmd-Shift-i), paste and run the following code in the console:
```js
$('.CodeMirror').CodeMirror.setOption('continueComments', null)
```

**Comment by tophf** [comment]
Also, <kbd>Shift</kbd><kbd>Enter</kbd> should break lines without continuation, which is probably a much simper workaround.

**Comment by tophf** [comment]
The next Violentmonkey will allow disabling it completely by setting custom CodeMirror options in advanced settings, for example:
```
{
  "continueComments": false,
  "lineWrapping": true,
  "indentWithTabs": false,
  "indentUnit": 2,
  "undoDepth": 200
}
```

**Comment by gera2ld** [comment]
fixed in v2.12.1
